### PR TITLE
disable set, enum and bit pushed down

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
This PR disables set, enum and bit pushed down. 

The reason we do that is because TiKV does not support these type's pushdown. 

This PR closes https://github.com/pingcap/tispark/issues/1244

### CODE REVIEW

**Comment by marsishandsome** [comment]
please add `why disables set, enum and bit pushed down?` in PR description

File: core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala [hunk 1]
```diff
@@ -75,10 +73,11 @@ object TiUtil {
 
     if (expr.children.isEmpty) {
       expr match {
-        // bit/duration type is not allowed to be pushed down
+        // bit, duration, set, and enum type is not allowed to be pushed down
         case attr: AttributeReference if nameTypeMap.contains(attr.name) =>
           val head = nameTypeMap.get(attr.name).head
           return !head.isInstanceOf[BitType]
+          return nameTypeMap.get(attr.name).head.isPushDownSupported
```
Comment (marsishandsome) [inline core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: two `returns`?

Comment (zhexuany) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: The second return is the real return. I'll fix it.

Comment (zhexuany) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: The second return is the real return. I'll fix it.

Comment (birdstorm) [reply core/src/main/scala/com/pingcap/tispark/utils/TiUtil.scala]: two replies and two returns... lmao

File: core/src/main/scala/org/apache/spark/sql/TiStrategy.scala [hunk 2]
```diff
@@ -308,8 +308,13 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
     sortOrder: Seq[SortOrder]
   ): SparkPlan = {
     val request = new TiDAGRequest(pushDownType(), timeZoneOffsetInSeconds())
-    request.setLimit(limit)
-    addSortOrder(request, sortOrder)
+    val filteredSortOrder =
+      sortOrder.filter(order => TiUtil.isSupportedOrderBy(order, source, blacklist))
+    if (filteredSortOrder.nonEmpty) {
```
Comment (birdstorm) [inline core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: does this mean that limit will not be pushed down if there isn't an `order by` statement? It seems quite strange.

Comment (zhexuany) [reply core/src/main/scala/org/apache/spark/sql/TiStrategy.scala]: yes.

File: tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java [hunk 3]
```diff
@@ -322,6 +322,13 @@ protected boolean isPrefixIndexSupported() {
    */
   public abstract Object getOriginDefaultValueNonNull(String value, long version);
 
+  /*
+   * return true if this type can be pushed down to TiKV or TiFLASH
+   * */
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java]: pls fix the format of the comment

**Comment by zhexuany** [comment]
`/run-all-tests

File: tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java (hunk 3 above)
Comment (zhexuany) [reply tikv-client/src/main/java/com/pingcap/tikv/types/DataType.java]: DONE

**Pull request merged**